### DISCUSSION TITLE
Learning Livewire and Laravel. Need help with Modal button

### DISCUSSION THREAD

**Discussion by amanvexed** [livewire/livewire#3943] [1. Help]
Hi everyone,

 I am learning Laravel and Livewire at the moment. I created a Modal that I want to use to capture data. The problem I have is that the wire:click that should be handled in the Component class when clicked is not working. I would appreciate guidance and help. My modal code and component class code is below. Thanks:

======Component Class============
```<?php
namespace App\Http\Livewire;
use Livewire\Component;

class AirtimeModal extends Component
{
    public $show = true;

    public function productName(){
          $this->show="false;
    }

    public function render()
    {
        return view('livewire.airtime-modal', ['data' => $this->show]);
    }
}```


========= Modal===========
```<div wire:ignore.self class="modal fade" id="airtime" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-header">
                <h5 class="modal-title" id="exampleModalLabel">BUY AIRTIME {{$show}}</h5>
                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                     <span aria-hidden="true close-btn">×</span>
                </button>
            </div>
            <div class="modal-body">
                <form>
                    <div class="modal-footer">
                        <button type="button" class="btn btn-secondary close-btn" data-dismiss="modal">Close</button>
                        <button type="button" wire:click="productName" class="btn btn-primary">Continue</button>
                           </div>
                </form>
            </div>
        </div>
    </div>
</div>```

**Comment by denizozturk** [comment]
Hi, you forgot double quotes here `$this->show="false;` This might be causing problems. Remove `"`, and I couldn't understand why you are using the `wire:ignore.self` property here. 

A little tip, if you have a button with a type value of submit for form operations, you can add `wire:submit.prevent="functionName"` to the form element.

More tips: https://laravel-livewire.com/docs/2.x/actions

**Reply by amanvexed** [reply]
I did all you said. I removed the wire:ignore.slef, updated $this->show=false, tested with wire:submit.prevent and it still didn't work. I seem to be doing something wrong. Because I removed the buttons from the modal and put it in the HTML file and the method was not triggered in the component class

**Reply by denizozturk** [reply]
> I did all you said. I removed the wire:ignore.slef, updated $this->show=false, tested with wire:submit.prevent and it still didn't work. I seem to be doing something wrong. Because I removed the buttons from the modal and put it in the HTML file and the method was not triggered in the component class

What are you using the show variable for? to hide the modal?

**Reply by amanvexed** [reply]
I rebuilt my project following the example in https://www.nicesnippets.com/blog/laravel-livewire-crud-with-bootstrap-modal-example and it started working. Thanks for taking time to help me.
